### PR TITLE
Enable wayland socket

### PR DESCRIPTION
**Pull request by bbhtt**
Regarding https://github.com/flathub/io.freetubeapp.FreeTube/issues/67

This enables the wayland socket by default, so that people can try out Electron's experimental wayland support. The x11 socket is still needed as it defaults to that, `fallback-x11` won't work.

This will possibly show a warning/fail due to the [linter](https://github.com/flathub/flatpak-builder-lint/blob/ff408c87c1ebefe6276609eb536464fd181de5f9/flatpak_builder_lint/checks/finish_args.py#L37), so an exception needs to be requested in a [similar fashion](https://github.com/flathub/flatpak-builder-lint/blob/ff408c87c1ebefe6276609eb536464fd181de5f9/flatpak_builder_lint/staticfiles/exceptions.json#L1190) if necessary.

To test native wayland: 

```sh
flatpak run --nosocket=x11 io.freetubeapp.FreeTube --ozone-platform-hint=auto --enable-features=WaylandWindowDecorations
```

### CODE REVIEW

**Comment by pm4rcin** [comment]
I have run it for many months in wlroots compositors. Sway and dwl work great with native wayland. What do you think about adding --ozone-platform-hint=auto to the executable so it picks whatever is available by default like here https://github.com/flathub/me.hyliu.fluentreader/commit/9be863dc55b10abacf72a8ad20690a3e8f9f4a47 which works perfectly (or at least nobody complaints) .

**Comment by pm4rcin** [comment]
I have run it for many months in wlroots compositors. Sway and dwl work great with native wayland. What do you think about adding --ozone-platform-hint=auto to the executable so it picks whatever is available by default like here https://github.com/flathub/me.hyliu.fluentreader/commit/9be863dc55b10abacf72a8ad20690a3e8f9f4a47 which works perfectly (or at least nobody complaints) ?

**Comment by bbhtt** [comment]
> I have run it for many months in wlroots compositors. Sway and dwl work great with native wayland. What do you think about adding --ozone-platform-hint=auto to the executable so it picks whatever is available by default like here [flathub/me.hyliu.fluentreader@9be863d](https://github.com/flathub/me.hyliu.fluentreader/commit/9be863dc55b10abacf72a8ad20690a3e8f9f4a47) which works perfectly (or at least nobody complaints) ?

I'd advise against it. Enabling something the Electron upstream doesn't have enabled by default or actively works on invites bug reports Flathub maintainers or app developers can't do anything about.

The native wayland mode in Electron in broken in many ways like scaling, popups etc. (look in the electron repo for 'wayland'). More recently Electron v25 broke window controls in apps https://github.com/electron/electron/issues/38786

In conclusion, it's not stable.

In the meantime, flathub now has a linter that prevents having both wayland and x11 sockets at the same time. So I'll have to close this.

Maybe the info to use wayland can be added in the readme here so that whoever wants to can do so? cc @PrestonN

**Comment by pm4rcin** [comment]
What about doing like signal flatpak did via env variable. Just look at [that](https://github.com/flathub/org.signal.Signal/blob/master/signal-desktop.sh). To me it seems like the best for now because by default it'll start on X11 but one can activate wayland by changing one value.

**Comment by bbhtt** [comment]
> What about doing like signal flatpak did via env variable? Just look at [that](https://github.com/flathub/org.signal.Signal/blob/master/signal-desktop.sh). To me it seems like the best for now because by default it'll start on X11 but one can activate wayland by changing one value.

Yea that might be doable, feel free to open a PR if you want to. But seems PrestoN is inactive for a while.
